More refactoring around boolean expression suggestion.

LaurentMazare opened the pull request:
Yet another PR on boolean expression suggestion.
As discussed previously, I moved some arguments of the recursive call in a dedicated struct.
I also noticed that the brackets argument was only set to true when calling `recurse` on `Or` or `And` in the `Not` branch or when calling `recurse` on `Or` in the `And` case. It's hence straightforward to remove this argument and have the brackets being added by the caller.
In particular, this removes the special case where `Term` gets bracketed only for binary expressions. I think this case was actually unused and the current behavior is correct as `snip` will contain some parenthesis if there were any in the original code.

oli-obk commented:
No clue what I was thinking with that bracket stuff.

> In particular, this removes the special case where Term gets bracketed only for binary expressions. I think this case was actually unused and the current behavior is correct as snip will contain some parenthesis if there were any in the original code.

Can you try adding a test for that?

LaurentMazare commented:
Sure, I already included a test with a xor in the PR and just added two variants to check.

oli-obk commented:
Lgtm. We hit a new nightly though.

LaurentMazare commented:
I updated the tests to match the new nightly (some '|' lines were modified). I also rebased my branch to the latest upstream from the main repo, not sure I did that properly as I know see lots of the commits of the main branch in my PR.

oli-obk commented:
> I also rebased my branch to the latest upstream from the main repo, not sure I did that properly as I know see lots of the commits of the main branch in my PR.

Doesn't look like it worked properly. Try `git rebase origin/master`

LaurentMazare commented:
Ok, I undoed the merge and hopefully it looks better now.

oli-obk commented:
Superb!
